pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -168,7 +172,18 @@ private SuccessResponse addOrUpdateSchema(@Nullable String schemaName, FormDataM
     }
 
     try {
+      Schema oldSchema;
+      if (schemaName == null) {
+        // New schema posted
+        oldSchema = null;
+      } else {
+        oldSchema = _pinotHelixResourceManager.getSchema(schemaName);
+      }
       _pinotHelixResourceManager.addOrUpdateSchema(schema);
+
+      LOGGER.info("Metadata change notification from old schema {} to new schema {}", oldSchema, schema);
```

Comment:
add a comment saying the notification is best effort. If controller fails at this point, no notification will be given.